### PR TITLE
Adding validations for transform

### PR DESCRIPTION
**Pull request by thalurur**
*Issue #, if available:*

*Description of changes:*
Adding validation for transforms and some tests

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/model/dimension/Dimension.kt [hunk 1]
```diff
@@ -40,6 +40,8 @@ abstract class Dimension(
 
     abstract fun toSourceBuilder(): CompositeValuesSourceBuilder<*>
 
+    abstract fun canBeRealizedInMappings(mappings: Map<String, Any>): Boolean
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/model/dimension/Dimension.kt:43]: Can you add a kdoc explaining what this means?

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformRunnerIT.kt [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+import org.elasticsearch.index.query.TermQueryBuilder
+import org.elasticsearch.script.Script
+import org.elasticsearch.script.ScriptType
+import org.elasticsearch.search.aggregations.AggregationBuilders
+import org.elasticsearch.search.aggregations.AggregatorFactories
+import org.elasticsearch.search.aggregations.metrics.ScriptedMetricAggregationBuilder
+
+class TransformRunnerIT : TransformRestTestCase() {
+
+    fun `test transforms`() {
```
Comment (dbbaughe) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformRunnerIT.kt:37]: Can't say I'm the biggest fan of smashing all the tests into a single test... 🤔

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            return TransformValidationResult(issues.isEmpty(), issues)
+        } catch (e: RemoteTransportException) {
+            val unwrappedException = ExceptionsHelper.unwrapCause(e) as Exception
+            throw TransformValidationException(errorMessage, unwrappedException)
+        } catch (e: Exception) {
+            throw TransformValidationException(errorMessage, e)
+        }
+    }
+
+    private suspend fun validateIndex(index: String, transform: Transform): List<String> {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt:59]: Add some kdocs to explain what this is doing

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import org.elasticsearch.transport.RemoteTransportException
+
+class TransformValidator(
+    private val indexNameExpressionResolver: IndexNameExpressionResolver,
+    private val clusterService: ClusterService,
+    private val esClient: Client
+) {
+
+    // TODO: When FGAC is supported in transform should check the user has the correct permissions
+    suspend fun validate(transform: Transform): TransformValidationResult {
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt:40]: Add some kdocs to explain what this is doing

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformRunnerIT.kt (hunk 2 above)
Comment (thalurur) [reply src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformRunnerIT.kt:37]: Sure, will move to different test methods

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt (hunk 3 above)
Comment (thalurur) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt:59]: will add

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt (hunk 4 above)
Comment (thalurur) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformValidator.kt:40]: will add

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/model/dimension/Dimension.kt (hunk 1 above)
Comment (thalurur) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/model/dimension/Dimension.kt:43]: will add

**Review by dbbaughe — APPROVED** [review]

**Pull request merged**